### ISSUE TITLE
Issue Owner via Tag

### ISSUE THREAD

**Issue by nao3301** [getsentry/sentry#8654]
I would love to set an issue Owner with a tag.

like Jone Doe ist the owner for all issues with the tag: 'frontend'

**Comment by dcramer** [comment]
This is something we've considered, but opted out to avoid complexity today.

Can you describe the use case a bit more? Are you using something like "serviceName=frontend" as a tag?

**Comment by nao3301** [comment]
yes, we have some microservices and some dev are responsible for some services.

**Comment by dleavitt** [comment]
Wanted to plus one this and suggest that what's needed here is a subset of the alert settings UI. From my (blurry, poorly-informed) perspective, owner assignment and alerting should be unified into "Rules" or at least share a similar interface. "When this happens, do this."

Use case (basically a long version @nao3301 said): 
* We have an application that depends on several underlying services. 
* Bug ownership is by service.
* Most of our endpoints use one or more of these services (so url is not a good assignment criterion.) 
* Service failures will cause errors in the endpoints consuming them (so file path is not a good assignment criterion.)
* Right now we have all our exceptions tagged by service, if only we could use these tags to assign ownership! :)

**Comment by n-peugnet** [comment]
Setting ownership via a tag filter would be indeed extremely useful in our case:

We report some errors from external sources via a single aws-lambda using `captureMessage` so there is no path or url we can filter from. On the other hand it is extremely easy for us to set a  tag to each message.

So filtering by tag would solve this problem using for example the following syntax:

```
tag(service):notifications n.peugnet@club1.fr
```

**Comment by caleb15** [comment]
I would suggest making the issue owner a tag and then allowing the ability to edit tags. A more generic solution so you don't have to write one-off functionality just for issue owner. This would also have the benefit of allowing people do send alerts based on issue owner.

**Comment by MoerkerkeLander** [comment]
Is there an update of this will every be implemented?

We would benefit of this kind of functionality due to the fact that all our microservices use the same template and a clear distinction between url or path is not valid per project.

**Comment by Finkes** [comment]
This would also be very helpful for us. In our case we have a central sentry project collecting issues from various sources. When we send the error to sentry we already add a tag which can be mapped to the owner 1:1. Is this something that you plan for the near future?

**Comment by untitaker** [comment]
I personally agree that this would be useful.

If you really need this *right now* you can work around this by inserting a fake frame in your stacktrace with a made-up filename/abspath that you then use to assign issue owners. You should be able to do this with the Python SDK like this:

```python
def before_send(event, hint):
    event.setdefault('stacktrace', {}).setdefault("frames", []).append({"abs_path": "issueowner://issue owner name here"})
    return event

init(..., before_send=before_send)
```

Of course this is really ugly and will also create new issues because of the new stacktrace. You can fix *that* using [server-side fingerprinting/grouping enhancements](https://docs.sentry.io/data-management/event-grouping/grouping-enhancements/):

```
path:issueowner://* -group
```

**Comment by mitsuhiko** [comment]
Fixed by #18459
